cmds/webboot/webboot.go

```diff
@@ -0,0 +1,245 @@
+	return u.isDefault
+}
+
+func (u ISO) String() string {
+	return fmt.Sprintf("%+v\n", u)
+}
+
+// Label is the string this iso displays in the menu page.
+func (u ISO) Label() string {
+	return u.name
+}
+
+// IsDefault is for mark whether this entry is a default choise. we assume download option always be a default option.
+func (u DownloadOption) IsDefault() bool {
+	return true
+}
+
+func (u DownloadOption) String() string {
+	return fmt.Sprintf("%+v\n", u)
+}
+
+// Label is the string this entry displays in the menu page. download option will be displayed as "Download" in the menu page.
+func (u DownloadOption) Label() string {
+	return "Download"
+}
+
+// getKernel is to find the kernel of a ISO. if the path of ISO is given, just check if the kernel is exist.
+func getKernel(u *ISO) error {
+
+	log.Println("try mount iso...")
+
+	isoName := u.name
+
+	diskFile, err := ioutil.TempDir("", "/mnt-iso")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(diskFile)
+
+	mountPath := filepath.Join(diskFile, isoName)
+	if mp, err := mount.Mount(u.path, mountPath, "iso9660", "", mount.ReadOnly); err != nil {
+		log.Printf("TryMount %s = %v, want nil", u.path, err)
+	} else {
+		log.Printf("mounted disk file is %s\n", mountPath)
+		// if kernel is given, check its validation
+		if u.kernel != "" {
+			walkfunc := func(path string, info os.FileInfo, err error) error {
+				if path == filepath.Join(mountPath, u.kernel) {
+					log.Printf("Kernel at %s is found\n", u.kernel)
+				}
+				return nil
+			}
+
+			log.Println("\nfinding kernel...")
+			filepath.Walk(diskFile, walkfunc)
+			log.Printf("Kernel path is %s\n", u.kernel)
+		}
+
+		log.Println("\ntry unmount iso...")
+		if err := mp.Unmount(0); err != nil {
+			log.Printf("Unmount(%q) = %v, want nil", mountPath, err)
+		}
+	}
+
+	log.Println("Done")
+	return nil
+}
+
+// isoMenu is to  build the iso menu page and return the iso path and kernel path of which user choose.
+func isoMenu(isos []ISO) (string, string) {
+
+	var entries []menus.Entry
+	for _, iso := range isos {
+		entries = append(entries, iso)
+	}
+	var downloadOption DownloadOption
+	entries = append(entries, downloadOption)
+
+	index, err := menus.DisplayMenu("ISO Menu", "Choose an iso you want to boot (hit enter to choose the default - 1 is the default option) >", 0, entries)
+
+	// if index is exceed the length of entries or less than 0 means something wrong happened
+	if err != nil || index < 0 || index >= len(entries) {
+		if err != nil {
+			log.Println(err)
+		}
+		return "", ""
+	}
+	// if index is the same as the length of entries - 1, it means the chosen option is download. Elsewise the chosen option is a cached iso.
+	if index == len(entries)-1 {
+		// if download option is chosen, webboot will show another input window for user to indicate what they want
+		isoLabel, err := menus.NewInputWindow("Please input the name of iso you want to download and install here:", 100, 1, func(input string) (bool, string) {
```

Comment:
Nit: Could you declare the func here as a separate variable and then pass that the menus.NewInputWindow so that this is a little easier to read?